From f63c5b7e7ddd3b0af569d4128bbeeee1161cee12 Mon Sep 17 00:00:00 2001
Message-Id: <f63c5b7e7ddd3b0af569d4128bbeeee1161cee12.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 30 Jul 2014 15:00:16 +0530
Subject: [PATCH 238/312] FOR_UPSTREAM [VPG]: Ungate Display Pipe A subsystem
 Control before accessing display register

On CHV, before accessing dp_aux_ch register or DPIO register, we need to make
sure "Display Pipe A subsystem Control", or else we will read a Junk data which
might fail display.

FIXME: The Punit in current hardware doesn't seem ready for this yet, so leave
it iffed out. As of now, PIPE A hardcode.

Change-Id: Ie8579c9ce3cda36cabccb5d0d5bdaeb8c2d97661
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    4 ++--
 drivers/gpu/drm/i915/intel_dp.c      |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 9c72c58..6a1e968 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5601,7 +5601,7 @@ void intel_connector_dpms(struct drm_connector *connector, int mode)
 
 	connector->dpms = mode;
 
-	intel_runtime_pm_get(dev_priv);
+	intel_display_power_get(dev_priv, PIPE_A);
 
 	if (mode == DRM_MODE_DPMS_ON)
 		intel_modeset_setup_hw_state(dev, true);
@@ -5613,7 +5613,7 @@ void intel_connector_dpms(struct drm_connector *connector, int mode)
 	if (mode == DRM_MODE_DPMS_ON)
 		intel_modeset_check_state(connector->dev);
 
-	intel_runtime_pm_put(dev_priv);
+	intel_display_power_put(dev_priv, PIPE_A);
 }
 
 /* Simple connector->get_hw_state implementation for encoders that support only
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 42f8ad7..1066a45 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -625,7 +625,7 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 
 	intel_dp_check_edp(intel_dp);
 
-	intel_aux_display_runtime_get(dev_priv);
+	intel_display_power_get(dev_priv, PIPE_A);
 
 	/* Try to wait for any previous AUX channel activity */
 	for (try = 0; try < 3; try++) {
@@ -719,7 +719,7 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 	ret = recv_bytes;
 out:
 	pm_qos_update_request(&dev_priv->pm_qos, PM_QOS_DEFAULT_VALUE);
-	intel_aux_display_runtime_put(dev_priv);
+	intel_display_power_put(dev_priv, PIPE_A);
 
 	if (vdd)
 		edp_panel_vdd_off(intel_dp, false);
-- 
1.7.9.5

